### PR TITLE
[FIX] WMS: cancel return should reassign move destination.

### PR DESCRIPTION
**Pull request by jir-odoo**
Task : https://www.odoo.com/web#id=12762&view_type=form&model=project.task&menu_id=3942&action=327

Pad : https://pad.odoo.com/p/r.rLOfjEuqxD2A0f8B

### CODE REVIEW

File: addons/stock/stock.py [hunk 1]
```diff
@@ -2292,16 +2292,17 @@ def action_cancel(self, cr, uid, ids, context=None):
...
                         self.action_cancel(cr, uid, [move.move_dest_id.id], context=context)
                     elif move.move_dest_id.state == 'waiting':
                         #If waiting, the chain will be broken and we are not sure if we can still wait for it (=> could take from stock instead)
                         self.write(cr, uid, [move.move_dest_id.id], {'state': 'confirmed'}, context=context)
+                elif move.origin_returned_move_id and move.origin_returned_move_id.move_dest_id:
+                    self.action_assign(cr, uid, [move.origin_returned_move_id.move_dest_id.id], context=context)
                 if move.procurement_id:
                     # Does the same as procurement check, only eliminating a refresh
                     procs_to_check.append(move.procurement_id.id)
-
```
Comment (tivisse) [inline addons/stock/stock.py]: Useless diff

**Comment by bst-odoo** [comment]
@jco-odoo Can you have a look at this PR ? 
Thank you.

File: addons/stock/stock.py [hunk 2]
```diff
@@ -2440,12 +2440,14 @@ def action_cancel(self, cr, uid, ids, context=None):
                     procurement_ids = procurement_obj.search(cr, uid, [('move_dest_id', '=', move.id)], context=context)
                     procurement_obj.cancel(cr, uid, procurement_ids, context=context)
             else:
-                if move.move_dest_id:
+                if move.move_dest_id and not move.origin_returned_move_id:
```
Comment (jco-odoo) [inline addons/stock/stock.py]: So, this means that a move that was returned, should not propagate its cancels: why?

File: addons/stock/stock.py [hunk 3]
```diff
@@ -2440,12 +2440,14 @@ def action_cancel(self, cr, uid, ids, context=None):
...
                     if move.propagate:
                         self.action_cancel(cr, uid, [move.move_dest_id.id], context=context)
                     elif move.move_dest_id.state == 'waiting':
                         #If waiting, the chain will be broken and we are not sure if we can still wait for it (=> could take from stock instead)
                         self.write(cr, uid, [move.move_dest_id.id], {'state': 'confirmed'}, context=context)
+                elif move.origin_returned_move_id and move.origin_returned_move_id.move_dest_id:
+                    self.action_assign(cr, uid, [move.origin_returned_move_id.move_dest_id.id], context=context)
```
Comment (jco-odoo) [inline addons/stock/stock.py]: I think what you wanted to do is move.origin_returned_move_id.origin_returned_move_id.move_dest_id

Comment (bst-odoo) [inline addons/stock/stock.py]: @jco-odoo is it ok now ?

Comment (bst-odoo) [inline addons/stock/stock.py]: up

**Pull request closed without merging**